### PR TITLE
Updates to Category page

### PR DESCRIPTION
**Pull request by jbt**
Further updates to follow on from #136 

 * Updated filter styles with collapsible sections
 * Updated product card styles to match products elsewhere
 * Updated grid layout to scale better at various responsive break points

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/web-ui/src/public/CategoryDetail.vue [hunk 1]
```diff
@@ -111,7 +122,9 @@ export default {
         intermediate = data
       }
 
-      if (this.personalizeUserID && intermediate.length > 0) {
+      const shouldPersonalise = false;
```
Comment (james-jory) [inline src/web-ui/src/public/CategoryDetail.vue:125]: Assuming this was added for testing? Please remove.

File: src/web-ui/src/public/components/Product.vue [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
-  <div class="col-sm-12 col-md-4 col-lg-3 mb-4">
+  <div class="featured-product p-3 d-flex flex-column justify-content-between text-left">
     <!-- Card -->
     <router-link class="link" :to="{name:'ProductDetail', params: {id: product.id}, query: {feature: feature, exp: experimentCorrelationId}}">
-      <div class="card mx-auto h-100 product-card text-left">
+      <div>
         <img :src="productImageURL" class="card-img-top" :alt="product.name">
-        <div class="card-body">
-          <h5 class="card-title">{{ product.name }}</h5>
-          <p class="card-text">{{ product.description }}</p>
```
Comment (james-jory) [inline src/web-ui/src/public/components/Product.vue:9]: I didn't realize that product description wasn't going to be used anymore. Is it used elsewhere in the app? Should it be removed from the data model completely?

File: src/web-ui/src/public/components/Product.vue [hunk 3]
```diff
@@ -1,18 +1,15 @@
...
-        </div>
-        <div class="card-footer" v-if="experiment">
-          <small class="text-muted"><i class="fa fa-balance-scale"></i> {{ experimentDescription }}</small>
-        </div>
+        <div class="product-name">{{ product.name }}</div>
+        <FiveStars />
+        <div>${{ product.price.toFixed(2) }}</div>
+      </div>
+      <div v-if="experiment" class="experiment mt-1 d-flex align-items-center text-muted">
+        <i class="fa fa-balance-scale mr-2"></i> {{ getExperimentDescription(experiment) }}
```
Comment (james-jory) [inline src/web-ui/src/public/components/Product.vue:12]: `experimentDescription` is a computed property. Does the call to `getExperimentDescription()` passing the experiment work? Did you test with an active experiment?

**Review by james-jory — CHANGES_REQUESTED** [review]

File: src/web-ui/src/public/components/Product.vue (hunk 2 above)
Comment (kazbaig) [reply src/web-ui/src/public/components/Product.vue:9]: Descriptions are still used on the product detail page.

The goal with the product cards on this page is to have the same look as the product carousels used throughout the app. We can probably refactor to use the same component in the future.

File: src/web-ui/src/public/CategoryDetail.vue [hunk 4]
```diff
@@ -1,50 +1,61 @@
...
-        <div class="row">
+        <div class="row mt-4">
 
           <div class="col-sm-3 col-md-3 col-lg-3 text-left">
-            <div class="card mb-3">
-              <div class="card-body">
-                <h5 class="card-title mb-0">Gender</h5>
+            <h4 class="bg-light p-2">Filters</h4>
+            <div class="mb-3 border-bottom">
+                <h5
```
Comment (kazbaig) [inline src/web-ui/src/public/CategoryDetail.vue:16]: Let's make the collapsable anchors buttons instead to maintain semantic markup.

Also, the Gender anchor doesn't have enough margin-bottom currently. It looks too close to the divider when collapsed as a consequence.

File: src/web-ui/src/public/CategoryDetail.vue [hunk 5]
```diff
@@ -193,4 +206,19 @@ export default {
   .carousel-item {
     max-height: 600px;
   }
+
+  .card-deck {
+    display: grid;
+    grid-gap: 1rem;
+    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr) ) ;
```
Comment (kazbaig) [inline src/web-ui/src/public/CategoryDetail.vue:213]: Looking a lot better now :)

File: src/web-ui/src/public/components/Product.vue (hunk 3 above)
Comment (kazbaig) [reply src/web-ui/src/public/components/Product.vue:12]: I think this was the result of some copy-paste from the `ProductCarousel` component. It should indeed be using `experimentDescription` here.

File: src/web-ui/src/public/CategoryDetail.vue (hunk 1 above)
Comment (jbt) [reply src/web-ui/src/public/CategoryDetail.vue:125]: Yep no idea how I let that slip through, removing now

**Comment by kazbaig** [comment]
LGTM!

**Review by james-jory — APPROVED** [review]

**Pull request merged**